code/game/objects/items/robot/robot_items/storage.dm

```diff
@@ -0,0 +1,213 @@
+
+/obj/item/borg/apparatus/beaker/Initialize(mapload)
+	. = ..()
+	add_glass()
+	RegisterSignal(stored, COMSIG_ATOM_UPDATED_ICON, .proc/on_stored_updated_icon)
+	update_appearance()
+
+/obj/item/borg/apparatus/beaker/Destroy()
+	if(stored)
+		var/obj/item/reagent_containers/container = stored
+		container.SplashReagents(get_turf(src))
+	QDEL_NULL(stored)
+	. = ..()
+
+/obj/item/borg/apparatus/beaker/proc/add_glass()
+	stored = new /obj/item/reagent_containers/glass/beaker/large(src)
+
+/obj/item/borg/apparatus/beaker/examine()
+	. = ..()
+	if(stored)
+		var/obj/item/reagent_containers/container = stored
+		. += "The apparatus currently has [container] secured, which contains:"
+		if(length(container.reagents.reagent_list))
+			for(var/datum/reagent/reagent in container.reagents.reagent_list)
+				. += "[reagent.volume] units of [reagent.name]"
+		else
+			. += "Nothing."
+
+		. += span_notice(" <i>Right-clicking</i> will splash the beaker on the ground.")
+	. += span_notice(" <i>Alt-click</i> will drop the currently stored beaker. ")
+
+/obj/item/borg/apparatus/beaker/update_overlays()
+	. = ..()
+	var/mutable_appearance/arm = mutable_appearance(icon = icon, icon_state = "borg_beaker_apparatus_arm")
+	if(stored)
+		COMPILE_OVERLAYS(stored)
+		stored.pixel_x = 0
+		stored.pixel_y = 0
+		var/mutable_appearance/stored_copy = new /mutable_appearance(stored)
+		if(istype(stored, /obj/item/reagent_containers/glass/beaker))
+			arm.pixel_y = arm.pixel_y - 3
+		stored_copy.layer = FLOAT_LAYER
+		stored_copy.plane = FLOAT_PLANE
+		. += stored_copy
+	else
+		arm.pixel_y = arm.pixel_y - 5
+	. += arm
+
+/// Secondary attack spills the content of the beaker.
+/obj/item/borg/apparatus/beaker/pre_attack_secondary(atom/target, mob/living/silicon/robot/user)
+	var/obj/item/reagent_containers/stored_beaker = stored
+	stored_beaker.SplashReagents(drop_location(user))
+	loc.visible_message(span_notice("[user] spills the contents of [stored_beaker] all over the ground."))
+	return ..()
+
+/obj/item/borg/apparatus/beaker/extra
+	name = "secondary beaker storage apparatus"
+	desc = "A supplementary beaker storage apparatus."
+
+/obj/item/borg/apparatus/beaker/service
+	name = "beverage storage apparatus"
+	desc = "A special apparatus for carrying drinks without spilling the contents. Will resynthesize any drinks you pour out!"
+	icon_state = "borg_beaker_apparatus"
+	storable = list(/obj/item/reagent_containers/food/drinks,
+					/obj/item/reagent_containers/food/condiment)
+
+/obj/item/borg/apparatus/beaker/service/add_glass()
+	stored = new /obj/item/reagent_containers/food/drinks/drinkingglass(src)
+	handle_refilling(stored, loc.loc, force = TRUE)
+
+/obj/item/borg/apparatus/beaker/service/proc/handle_refilling(obj/item/reagent_containers/glass, mob/living/silicon/robot/bro, force = FALSE)
+	if (isnull(bro))
+		bro = loc
+	if (!iscyborg(bro))
+		return
+
+	if (!stored || force)
+		glass.AddComponent(/datum/component/reagent_refiller, cell = bro?.cell)
```

Comment:
If borg cell is replaced won't this stop consuming power ?

Reply:
hmmm... yes it would